ProjectLighthouse/Files/FileHelper.cs

```diff
@@ -50,6 +51,50 @@ public static bool IsFileSafe(LbpFile file)
         };
     }
 
+    public static GameVersion ParseLevelVersion(LbpFile file)
+    {
+        if (file.FileType != LbpFileType.Level)
+        {
+            return GameVersion.Unknown;
+        }
+
+        // Revision numbers borrowed from https://github.com/ennuo/toolkit/blob/main/src/main/java/ennuo/craftworld/resources/structs/Revision.java
+        // LBP3 latest revision is 0x21803F9
+        // LBP2 latest revision is 0x3F8
+        // LBP1 latest revision is 0x272
+        // LBP Vita latest is 0x3E2
+        // There are like 1600 revisions so this doesn't cover everything
+        uint revision = 0;
+
+        // construct a 32 bit number from 4 individual bytes
+        for (int i = 4; i <= 7; i++)
+        {
+            revision <<= 8;
+            revision |= file.Data[i];
+        }
+        // construct a 16 bit number from 2 individual bytes
+        ushort branchId = (ushort) (file.Data[0xC] << 8 | file.Data[0xD]);
```

Comment:
Decimal numbers are a little more clear here imo
```suggestion
        ushort branchId = (ushort) (file.Data[12] << 8 | file.Data[13]);
```